api/src/main/java/org/openmrs/api/PatientService.java

```diff
@@ -844,4 +844,27 @@ public void saveCauseOfDeathObs(Patient patient, Date dateDied, Concept causeOfD
 	 * @throws PatientIdentifierTypeLockedException
 	 */
 	public void checkIfPatientIdentifierTypesAreLocked() throws PatientIdentifierTypeLockedException;
+	
+	/**
+	 * Adds actions to be performed before and after merging patients
+	 * 
+	 * @param patientMergeActions merge actions to be registered
+	 */
+	public void addPatientMergeAction(PatientMergeAction patientMergeAction);
+	
+	/**
+	 * Removes a registered merge action
+	 * 
+	 * @param patientMergeAction merge action to be removed
+	 */
+	public void removePatientMergeAction(PatientMergeAction patientMergeAction);
+	
+	public void setPatientMergeActions(List<PatientMergeAction> patientMergeActions);
+	
+	/**
+	 * Gets the patientmergeactions
+	 * 
+	 * @return returns a list of registered patient merge actions
+	 */
+	public List<PatientMergeAction> getPatientMergeActions();
```

Comment:
I wouldn't expose these getters/setters for merge actions, this should be internal logic to the service that should not be exposed.

Reply:
I was sort of not finding a better way to write tests  for it using the patientservice interface so i added it. Thanks. I think i really can do better than that.